src/coreclr/jit/gentree.cpp

```diff
@@ -10710,6 +10709,57 @@ void GenTree::SetIndirExceptionFlags(Compiler* comp)
     return charsDisplayed;
 }
 
+/* static */ int GenTree::gtDispHandleKind(GenTreeFlags flags)
+{
+    GenTreeFlags handleKind = flags & GTF_ICON_HDL_MASK;
+    switch (handleKind)
+    {
+        case 0:
+            return 0;
+        case GTF_ICON_SCOPE_HDL:
+            return printf("GTF_ICON_SCOPE_HDL");
+        case GTF_ICON_CLASS_HDL:
+            return printf("GTF_ICON_CLASS_HDL");
+        case GTF_ICON_METHOD_HDL:
+            return printf("GTF_ICON_METHOD_HDL");
+        case GTF_ICON_FIELD_HDL:
+            return printf("GTF_ICON_FIELD_HDL");
+        case GTF_ICON_STATIC_HDL:
+            return printf("GTF_ICON_STATIC_HDL");
+        case GTF_ICON_STR_HDL:
+            return printf("GTF_ICON_STR_HDL");
+        case GTF_ICON_OBJ_HDL:
+            return printf("GTF_ICON_OBJ_HDL");
+        case GTF_ICON_CONST_PTR:
+            return printf("GTF_ICON_CONST_PTR");
+        case GTF_ICON_GLOBAL_PTR:
+            return printf("GTF_ICON_GLOBAL_PTR");
+        case GTF_ICON_VARG_HDL:
+            return printf("GTF_ICON_VARG_HDL");
+        case GTF_ICON_PINVKI_HDL:
+            return printf("GTF_ICON_PINVKI_HDL");
+        case GTF_ICON_TOKEN_HDL:
+            return printf("GTF_ICON_TOKEN_HDL");
+        case GTF_ICON_TLS_HDL:
+            return printf("GTF_ICON_TLS_HDL");
+        case GTF_ICON_FTN_ADDR:
+            return printf("GTF_ICON_FTN_ADDR");
+        case GTF_ICON_CIDMID_HDL:
+            return printf("GTF_ICON_CIDMID_HDL");
+        case GTF_ICON_BBC_PTR:
+            return printf("GTF_ICON_BBC_PTR");
+        case GTF_ICON_STATIC_BOX_PTR:
+            return printf("GTF_ICON_STATIC_BOX_PTR");
+        case GTF_ICON_FIELD_SEQ:
+            return printf("GTF_ICON_FIELD_SEQ");
+        case GTF_ICON_STATIC_ADDR_PTR:
+            return printf("GTF_ICON_STATIC_ADDR_PTR");
+        default:
+            assert(!"a forgotten handle flag");
```

Comment:
Maybe not assert here? 

Slight preference for having methods like this just return strings, so they compose better into formatted output.

Also: we have both "Disp" and "dsp", should we settle on just one?

Reply:
> Maybe not assert here?
> 
> Slight preference for having methods like this just return strings, so they compose better into formatted output.

Agreed, and fixed.

> Also: we have both "Disp" and "dsp", should we settle on just one?

Maybe? IMO, slight preference for "disp". Not sure it's worth the effort to change it.